modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -118,6 +122,18 @@ public GeoJSONReader(URL url) throws IOException {
         baseName = FilenameUtils.getBaseName(url.getPath());
     }
 
+    /**
+     * Builds a GeoJSON parser from a GeoJSON source, located at the specified URL.
+     *
+     * @param url String representation of the URL
+     * @param idPrefix prefix to use for feature ids
+     * @throws IOException if the URL is not valid
+     */
+    public GeoJSONReader(URL url, String idPrefix) throws IOException {
+        this(url);
+        this.idPrefix = idPrefix;
```

Comment:
don't you need to call `this(url)` here or initialize the `parser`?

Reply:
See comment below about the featurebuilder name and the prefix.  If a seperate prefix is not needed, this new constructor will also go away.